nilearn/image/image.py

```diff
@@ -596,3 +599,80 @@ def new_img_like(ref_niimg, data, affine=None, copy_header=False):
         header['cal_max'] = np.max(data) if data.size > 0 else 0.
         header['cal_max'] = np.min(data) if data.size > 0 else 0.
     return ref_niimg.__class__(data, affine, header=header)
+
+
+def threshold_img(img, threshold, mask_img=None,
+                  thresholding_strategy='img_value'):
+    """ Thresholds the given input image based on specific strategy.
+
+    Parameters
+    ----------
+    img: a 3D/4D Niimg-like object
+        Image contains of statistical or atlas maps which should be thresholded.
+
+    threshold: float or string or number
+        If float, we consider it as intensity value and therefore will be used
+        to directly threshold the image meaning thresholding is based on image
+        intensities. The given value should be within the range of minimum and
+        maximum intensity of the image.
+        float option is used with thresholding_strategy='img_value'.
+        If string, it should finish with percent sign e.g. "80%" and
+        should be within the range of "0%" to "100%".
+        If number, we consider it as percentage in real number and should be within
+        the range between 0 and 100 same like string case.
+        Both string and real number options are used with thresholding_strategy='percentile'.
+
+    thresholding_strategy: string {'percentile', 'img_value'}, \
+        default 'img_value', optional
+        If 'percentile', image is thresholded based on the percentage of the
+        score on the image data. The scores which are survived above this
+        percentile are kept.
+        If 'img_value', voxels which have intensities greater than given float
+        value are kept.
+
+    mask_img: Niimg-like object, default None, optional
+        Mask image applied to mask the input data.
+        If None, no masking will be applied.
+
+    Returns
+    -------
+    threshold_img: Niimg-like object
+        thresholded image of the given input image.
+    """
+    from . import resampling
+    from .. import masking
+
+    img = check_niimg(img)
+    img_data = img.get_data()
```

Comment:
The data is modified inplace after. Should we have a 'copy' here?

Also, maybe we should use a nilearn._utils.niimg._safe_get_data to avoid a side effect.